lib/spack/spack/environment/environment.py

```diff
@@ -1024,6 +1024,47 @@ def add(self, user_spec, list_name=user_speclist_name):
 
         return bool(not existing)
 
+    def change_existing_spec(self, change_spec, list_name=user_speclist_name, match_spec=None):
+        """
+        Provide a single spec and assume you want to find another spec with the
+        same name and override it with any conflicting details in change_spec.
+
+        This is only allowed for environments that do not use matrices (i.e.
+        they have a single, explicitly-enumerated list of specs called
+        'specs').
+        """
+        if not (change_spec.name or (match_spec and match_spec.name)):
+            raise ValueError(
+                "Must specify a spec name to identify a single spec"
```

Comment:
does it have to match exactly one spec?

Reply:
I think this is resolved by https://github.com/spack/spack/pull/31995#discussion_r952303271